problem/0002-isolated-integration-tests.md

```diff
@@ -0,0 +1,102 @@
+- Contribution Name: Isolated Integration Tests
+- Implementation Owner: @kleimkuhler, @alpeb
+- Start Date: 2020-03-18
+- Target Date: 2020-04-03
+- RFC PR: [linkerd/rfc#12](https://github.com/linkerd/rfc/pull/12)
+- Linkerd Issue: [linkerd/linkerd2#4190](https://github.com/linkerd/linkerd2/issues/4190)
+- Reviewers: @grampelberg
+
+# Summary
+
+[summary]: #summary
+
+It should be easier to run integration tests in a more isolated way. Instead
+of the default testflow which isolates tests by namespace, tests will be
+isolated by cluster. With each test creating it's own cluster, it will ensure
+there is no existing Linkerd resources when starting the test. It will also
+allow tests to be run in parallel because each test can run within its own
+cluster.
+
+# Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+The default testflow for running integration tests isolates tests by namespace
+on a single cluster. This is problematic for the following reasons:
+- The resources from any previous or existing Linkerd installation must be
+  deleted in order to not conflict with the new installation.
+- Tests cannot be run in parallel because only a single control plane can be
+  installed on a cluster
+
+Once the change is implemented, it will be easier to isolate integration tests
+by more than just namespace. This will allow for each integration test to run
+independently of any previous run or existing installation.
+
+Additionally, with finer grained integration test isolation, it will be
+possible for the default testflow to run integration test in parallel. This
+will allow for a faster test runs.
+
+# Design proposal (Step 2)
+
+[design-proposal]: #design-proposal
+
+**Note**: This should be completed as part of `Step 2`.
+
+This is the technical portion of the RFC. Explain the design in sufficient
+detail that:
+
+- Its interaction with other features is clear
+- It is reasonably clear how the contribution would be implemented
+- Corner cases are dissected by example
+- Dependencies on libraries, tools, projects or work that isn't yet complete
+- Use Cases
+- Goals
+- Non-Goals
+- Deliverables
+
+# Prior art
+
+[prior-art]: #prior-art
```

Comment:
I think the intention of this section is to refer to third-party solutions we might look into to use, leverage or get inspired on  to solve the problems stated in this RFC. E.g. does Prow use this approach to parallel testing?

Maybe the deficiencies you listed here should be extracted into the problem-statement section above.